dbms/src/Storages/MergeTree/ReplicatedMergeTreeQuorumWriter.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include <unordered_map>
+#include <IO/ReadBuffer.h>
+#include <IO/ReadBufferFromString.h>
+#include <IO/WriteBuffer.h>
+#include <IO/WriteBufferFromString.h>
+#include <IO/ReadHelpers.h>
+
+#include <Storages/MergeTree/MergeTreeDataPart.h>
+
+namespace DB
+{
+
+struct ReplicatedMergeTreeQuorumWriter
+{
+    using PartitionIdToMaxBlock = std::unordered_map<String, Int64>;
+    using PartitonIdToPartName= std::unordered_map<String, String>;
+
+    PartitonIdToPartName added_parts;
+
+    MergeTreeDataFormatVersion format_version;
+
+    ReplicatedMergeTreeQuorumWriter(const std::string & old_added_parts, MergeTreeDataFormatVersion format_version_)
+    : format_version(format_version_)
+    {
+        read(old_added_parts);
+    }
+
+    /// Write new parts in string with added parts.
+    std::string write(const std::string & part_name)
```

Comment:
The `write` method should write to the WriteBuffer (and `toString` method should return a string).
The `read` method should read from the ReadBuffer (and `fromString` method should parse a string).